code/code/sys/sstring.cc

```diff
@@ -870,3 +870,11 @@ sstring sstring::escapeJson() const {
   }
   return ss.str();
 }
+
+/* 
+  Compares <this> with <other>, ignoring case, and returns same 
+  values as std::string::compare()
+*/
+int sstring::compareCaseless(const sstring &other) const {
+  return this->lower().compare(other.lower());
```

Comment:
There are languages where there are several lowercase options for the same uppercase variant. It's safer to convert to uppercase instead. Not that it matters a bit - we only support ASCII anyway.